# Mission Config of the sub
# ALL UNITS IN M

# manager_info docs
# timeout_secs: # of seconds before timing out, if 0 the task won't timeout
# outcome_* : how the manager should respond when a task exits with this outcome
#            skip : move on to the next task
#            later : move this task to the end of the mission_tasks queue, essentially coming back to the task later
#            uh_oh : or any unrecognized string will crash and stop the SM

using_sim_params : True
manager_task_status_pub_rate: 1

startup_delay: 0

tasks:
  startup:
    manager_info:
      timeout_secs: 5
    depth: -1.5
    dive_time: 10

  start_gate:
    manager_info:
      timeout_secs: 0
      outcome_not_found: 'skip' # skip, later
      outcome_timed_out: 'uh_oh'
    seconds_in_position: 2
    new_dv_check_rate: 30
    mag_target: 390
    center_pole_target: 200
    elev_target: 0.0
    elev_thresh: 0.05
    slay_carrot_dist: 1
    do_style: True
    spin_carrot_rads: 0.7
    strafe_right: false
    strafe_carrot: 0.008

  droppers:
    manager_info:
      timeout_secs: 0
      outcome_not_found: 'skip' # skip, later
      outcome_timed_out: 'skip'
    drop_depth: -2    
    seconds_in_position: 2 # how many seconds we need to be directly above droppers before diving
    xy_dist_thresh_app: 0.5 # how close in xy we need to be before diving
    xy_dist_thresh_drop: 0.1 # how close in xy we need to be before dropping


  jiangshi:
    manager_info:
      timeout_secs: 0
      outcome_not_found: 'skip' # skip, later
      outcome_timed_out: 'skip'
    seconds_in_position: 2
    new_dv_check_rate: 30
    mag_target: 152600 # target dvector magnitude
    elev_target: 0.0
    elev_thresh: 0.05 # how close elevation has to be
    slay_carrot_dist: 1 # how far to go ahead to hit jiangshi

mission_tasks: # Tasks To Do In Order
  # - startup
  - start_gate
  # - jiangshi
  # - droppers
  # - triangle
  
  # - octagon
